### PR TITLE
Better logging, unified methods for updating the DB

### PR DESCRIPTION
**Pull request by DennisHub**

### CODE REVIEW

File: database_mockup/src/animalkeeping/ui/PersonsView.java [hunk 1]
```diff
@@ -68,6 +68,19 @@ public void initialize(URL location, ResourceBundle resources) {
...
+        subjectCol.setCellValueFactory(data -> new ReadOnlyStringWrapper(data.getValue().getSubject().getName()));
+        typeCol = new TableColumn<>("treatment");
+        typeCol.setCellValueFactory(data -> new ReadOnlyStringWrapper(data.getValue().getTreatmentType().getName()));
+        startDateCol = new TableColumn<>("start");
+        startDateCol.setCellValueFactory(data -> new ReadOnlyObjectWrapper<>(data.getValue().getStart()));
+        endDateCol = new TableColumn<>("end");
+        endDateCol.setCellValueFactory(data -> new ReadOnlyObjectWrapper<>(data.getValue().getEnd()));
+        treatmentTable.getColumns().clear();
+        treatmentTable.getColumns().addAll(idCol, subjectCol, typeCol, startDateCol, endDateCol);
+
```
Comment (jgrewe) [inline database_mockup/src/animalkeeping/ui/PersonsView.java]: These additions result from a wrong merge?! with pr #25 this table was not created here by hand but uses the TreatmentTypeTable class

File: migrations/0002-changelog.sql [hunk 2]
```diff
@@ -7,5 +7,6 @@ CREATE TABLE IF NOT EXISTS `animal_keeping`.`changelog` (
   `created_time` DATETIME NOT NULL,
   `target_type` VARCHAR(200) NOT NULL,
   `target_id` INT NOT NULL, 
+  `who` VARCHAR(200) NOT NULL,
```
Comment (jgrewe) [inline migrations/0002-changelog.sql:10]: This change needs to go into a separate migration script. When we patch old migrations we will not be able to simply go on from an existing point (e.g. migration 0004 has already been applied) 
Just add a new script 0005-addusertochangelog_or_whatever_name.sql that contains the change:
````
ALTER TABLE ` animal_keeping`.`changelog` ADD `who` VARCHAR(200) NOT NULL;
```

**Comment by jgrewe** [comment]
the communicator is cool, removes lots of junk from the classes :)

File: database_mockup/src/animalkeeping/ui/PersonsView.java (hunk 1 above)
Comment (DennisHub) [reply database_mockup/src/animalkeeping/ui/PersonsView.java]: Possibly...?

File: migrations/0002-changelog.sql (hunk 2 above)
Comment (DennisHub) [reply migrations/0002-changelog.sql:10]: Done.

File: database_mockup/src/animalkeeping/ui/PersonsView.java (hunk 1 above)
Comment (DennisHub) [reply database_mockup/src/animalkeeping/ui/PersonsView.java]: Fixed.

**Pull request merged**